src/tally.rs

```diff
@@ -0,0 +1,265 @@
+        if let Err(ref e) = res
+        {
+            // Just try again if we have too many files open
+            if e.raw_os_error()
+                .ok_or_else(|| AocError::RunError("Error getting os error".into()))?
+                == TOO_MANY_OPEN_FILES
+            {
+                let dir = day.0.clone();
+                let target = target.clone();
+                set.spawn(async move {
+                    tokio::process::Command::new(target).current_dir(dir).output().await
+                });
+                continue;
+            }
+        }
+
+        let res = res?;
+        if !res.status.success()
+        {
+            return Err(AocError::RunError(format!("Error running day {}", day.1)));
+        }
+        vec.push(parse_get_times(res)?);
+    }
+
+    let len = vec.len();
+    let (p1, p2) = vec.into_iter().fold((0, Option::<usize>::None), |(p1, p2), (a, b)| {
+        (p1 + a, match (p2, b)
+        {
+            (Some(a), Some(b)) => Some(a + b),
+            (None, Some(b)) => Some(b),
+            _ => None,
+        })
+    });
+
+    Ok((p1 / len, p2.map(|val| val / len)))
+}
+
+async fn run_days(
+    days: Vec<(PathBuf, u32)>,
+    number_of_runs: usize,
+) -> Result<Vec<(u32, (usize, Option<usize>))>, AocError>
+{
+    let mut set = JoinSet::new();
+    let mut vec: Vec<(u32, (usize, Option<usize>))> = Vec::new();
+    let cargo_folder = cargo_path().await?;
+    for day in days
+    {
+        let cargo_folder = cargo_folder.clone();
+        set.spawn(async move { (day.1, run_day(cargo_folder, day, number_of_runs).await) });
+    }
+
+    while let Some(Ok((day, res))) = set.join_next().await
+    {
+        vec.push((day, res?));
+    }
+
+    Ok(vec)
+}
+
+fn print_info(days: Vec<(u32, (usize, Option<usize>))>, not_done: Vec<u32>, number_of_runs: usize)
+{
+    let red_text = |s: u32| format!("\x1b[0;33;31m{}\x1b[0m", s);
+    let gold_text = |s: &str| format!("\x1b[0;33;10m{}\x1b[0m:", s);
+    let silver_text = |s: &str| format!("\x1b[0;34;34m{}\x1b[0m:", s);
+
+    if !not_done.is_empty()
+    {
+        let mut not_done = not_done;
+        not_done.sort_unstable();
+        let mut s = String::new();
+        let mut first = true;
+        for day in not_done
+        {
+            if !first
+            {
+                s.push_str(", ");
+            }
+            s.push_str(&red_text(day));
+            first = false;
+        }
+        println!("Days not completed: {}", s);
+    }
+    println!("STATS:");
+    println!("Number of runs: {}:\n", number_of_runs);
+
+    let print_info = |text: String, vec: Vec<(u32, usize)>| {
+        println!("{}", text);
+
+        let mut _vec: Vec<_> = vec.iter().map(|(_, time)| *time).collect();
+        _vec.sort_unstable();
+
+        let median = _vec[_vec.len() / 2];
+
+        let total = vec.iter().map(|(_, time)| time).sum::<usize>();
+        let avg = total / vec.len();
+
+        let (day, time) = vec.iter().max_by_key(|k| k.1).unwrap();
```

Comment:
Nitpick: Maybe name this `highest` or `highest_time` or something to indicate it's the highest at least